nevergrad/optimization/es.py

```diff
@@ -64,13 +73,21 @@ def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None
             else:
                 self._waiting.append(candidate)
             if len(self._waiting) >= self._config.offsprings:
-                choices = self._waiting + ([] if self._config.only_offsprings else list(self._population.values()))
-                choices.sort(key=lambda x: x._meta["value"])
-                self._population = {x.uid: x for x in choices[:self._config.popsize]}
-                self._uid_queue.clear()
-                self._waiting.clear()
-                for uid in self._population:
-                    self._uid_queue.tell(uid)
+                self._simple_selection()
+
+
+    def _simple_selection(self):
+        choices = self._waiting + ([] if self._config.only_offsprings else list(self._population.values()))
+        if self._ranker is not None:
+            choices_rank = self._ranker.rank(choices, self._config.popsize)
+            choices.sort(key=lambda x: choices_rank[x.uid][0] if x.uid in choices_rank else float('inf'))
```

Comment:
why do we need to sort? this looks like a selection already, since it takes the popsize as input. The name should be updated accordingly

Reply:
Line 83 has been replaced by extracting the choices from the result choices_rank.